src/main/java/com/keboola/tableexporter/CsvWriter.java

```diff
@@ -0,0 +1,106 @@
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.sql.Clob;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.Iterator;
+
+public class CsvWriter {
+
+    private CSVPrinter csvPrinter;
+    private BufferedWriter writer;
+
+    public CsvWriter(String outputFile, String[] header) throws CsvException {
+        try {
+            writer = Files.newBufferedWriter(Paths.get(outputFile));
+            if (header != null) {
+                csvPrinter = new CSVPrinter(writer, CSVFormat.RFC4180.withHeader(header).withQuoteMode(QuoteMode.ALL));
+            } else {
+                csvPrinter = new CSVPrinter(writer, CSVFormat.RFC4180);
+            }
+        } catch (IOException e) {
+            throw new CsvException("Failed opening output file.", e);
+        }
+    }
+
+    public void write(ResultSet resultSet, boolean hasLobs) throws UserException, CsvException {
+        try {
+            if (hasLobs) {
+                this.printRecordsWithLobs(resultSet);
+            } else {
+                csvPrinter.printRecords(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new UserException("Failed to write csv ", e);
+        } catch (IOException e) {
+            throw new UserException("Failed to write csv ", e);
+        }
+    }
+
+    public void printRecordsWithLobs(ResultSet resultSet) throws SQLException, IOException, CsvException {
+        ResultSetMetaData rsMeta = resultSet.getMetaData();
+        int columnCount = rsMeta.getColumnCount();
+
+        while (resultSet.next()) {
+            for (int i = 1; i <= columnCount; ++i) {
+                if (rsMeta.getColumnTypeName(i) == "CLOB") {
+                    Clob clob = resultSet.getClob(i);
+                    String clobString = this.clobToString(clob);
+                    csvPrinter.print(clobString);
+                } else {
+                    csvPrinter.print(resultSet.getObject(i));
+                }
+            }
+            csvPrinter.println();
+        }
+    }
+
+    public void close() throws CsvException {
+        try {
+            csvPrinter.flush();
+            csvPrinter.close();
+            writer.close();
+        } catch (IOException e) {
+            throw new CsvException("Failed closing file writer", e);
+        }
+    }
+
+    private String clobToString(Clob clob) throws CsvException {
+        Reader clobReader = null;
+        try {
+            clobReader = clob.getCharacterStream();
+            final int clobLength = (int) clob.length();
+            char[] charBuffer = new char[clobLength];
+            int offset = 0;
+            int charsRead = clobReader.read(charBuffer, offset, clobLength);
+            while (charsRead > 0) {
+                offset += charsRead;
+                charsRead = clobReader.read(charBuffer, offset, clobLength);
+            }
```

Comment:
looking at this fresh, it looks bogus, I think it will break for long clobs.
first of all the string buffer has length `clob.length` (cast from long to int which is its own issue), and then in the while loop we keep reading into `charBuffer` from offset to clobLength.  So if this loop ever executes it will I think overflow the buffer.

Anyhow, I'm going to revisit the source in apache-commons convert